service/gcs/core/gcs/storage.go

```diff
@@ -31,51 +32,102 @@ const (
 	mappedDiskMountTimeout = time.Second * 2
 )
 
-// getLayerDevices turns a list of SCSI codes (0-0, 0-1, etc.) into a list of
-// the devices (sda, sdb, etc.) that they correspond to.
-func (c *gcsCore) getLayerDevices(scratch string, layers []prot.Layer) (scratchDevice string, layerDevices []string, err error) {
-	layerDevices = make([]string, len(layers))
+type mountSpec struct {
+	Source     string
+	FileSystem string
+	Flags      uintptr
+	Options    []string
+}
+
+const (
+	// From mount(8): Don't load the journal on mounting.  Note that if the
+	// filesystem was not unmounted cleanly, skipping the journal replay will
+	// lead to the filesystem containing inconsistencies that can lead to any
+	// number of problems.
+	mountOptionNoLoad = "noload"
+	// Enable DAX mode. This turns off the local cache for the file system and
+	// accesses the storage directly from host memory, reducing memory use
+	// and increasing sharing across VMs. Only supported on vPMEM devices.
+	mountOptionDax = "dax"
+
+	// For now the file system is hard-coded
+	defaultFileSystem = "ext4"
+)
+
+// Mount mounts the file system to the specified target.
+func (ms *mountSpec) Mount(osl oslayer.OS, target string) error {
+	options := strings.Join(ms.Options, ",")
+	err := osl.Mount(ms.Source, target, ms.FileSystem, ms.Flags, options)
```

Comment:
Ah. That's now a test bug -- I want mountSpec to contain the full path to the device, since there's no reason that the source needs to actually start with /dev/.